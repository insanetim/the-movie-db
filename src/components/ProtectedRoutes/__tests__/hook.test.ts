import { renderHook } from '@testing-library/react'

import useContainer from '../hook'

jest.mock('src/store/session/actions')

jest.mock('src/store/session/selectors', () => ({
  sessionIdSelector: jest.fn(() => 'test/sessionId')
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn().mockImplementation(() => ({}))
}))

describe('ProtectedRoutes useContainer hook', () => {
  it('should match snapshot', () => {
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })
})
