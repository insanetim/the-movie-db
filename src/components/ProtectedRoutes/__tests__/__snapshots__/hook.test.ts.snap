// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoutes useContainer hook should match snapshot 1`] = `
{
  "location": {},
  "sessionId": "test/sessionId",
}
`;
